pkg/resource/role/hooks.go

```diff
@@ -226,7 +226,13 @@ func (rm *resourceManager) syncInlinePolicies(
 			return err
 		}
 	}
+
 	for _, pair := range toDelete {
+		// do not remove elements we just updated with `addInlinePolicy`
+		if _, ok := lo.Find(toAdd, func(entry lo.Entry[string, string]) bool { return entry.Key == pair.Key }); ok {
```

Comment:
We want to use `lo.Find` because we only want to match entries using only their `.Key` field. `lo.Contains` does a match using both of `.Key` and `.Value`, which is not what we need in this case.